tests/dynamic_data/publications/layer_geoserver_proxy/geoserver_proxy_test.py

```diff
@@ -0,0 +1,140 @@
+        },
+    },
+    'private_layer_generic_url_no_user': {
+        'layer': PRIVATE_LAYER,
+        'wfs_params': {
+        },
+    },
+    'private_layer_generic_url_fraud_header': {
+        'layer': PRIVATE_LAYER,
+        'wfs_params': {
+            'request_headers': {settings.LAYMAN_GS_AUTHN_HTTP_HEADER_ATTRIBUTE: USER, },
+        },
+    },
+}
+
+
+@pytest.mark.usefixtures('oauth2_provider_mock')
+class TestWfsProxyOkCall(base_test.TestSingleRestPublication):
+    workspace = WORKSPACE
+    publication_type = process_client.LAYER_TYPE
+    rest_parametrization = []
+    usernames_to_reserve = [USER, USER_2]
+
+    test_cases = [
+        base_test.TestCaseType(
+            key=key,
+            publication=params['layer'],
+            type=EnumTestTypes.MANDATORY,
+            params=copy.deepcopy(params['wfs_params']),
+        ) for key, params in OK_TEST_CASES.items()
+    ]
+
+    def before_class(self):
+        self.post_publication(PUBLIC_LAYER, args=publications.SMALL_LAYER.definition, scope='class')
+        self.post_publication(PRIVATE_LAYER, args={'actor_name': USER, **publications.SMALL_LAYER.definition}, scope='class',)
+
+    def test_wfs_proxy(self, layer: Publication4Test, params):
+        wfs_client.post_wfst(*layer,
+                             operation=wfs_client.WfstOperation.INSERT,
+                             version=wfs_client.WfstVersion.WFS20,
+                             **params,
+                             )
+        assert_util.is_publication_valid_and_complete(layer)
+
+
+@pytest.mark.usefixtures('oauth2_provider_mock')
+class TestWfsProxyErrorCall(base_test.TestSingleRestPublication):
+    workspace = USER
+    publication_type = process_client.LAYER_TYPE
+    rest_parametrization = []
+    usernames_to_reserve = [USER, USER_2]
+
+    test_cases = [
+        base_test.TestCaseType(
+            key=key,
+            publication=params['layer'],
+            type=EnumTestTypes.MANDATORY,
+            params=copy.deepcopy(params['wfs_params']),
+        ) for key, params in ERROR_TEST_CASES.items()
+    ]
+
+    def before_class(self):
+        self.post_publication(PRIVATE_LAYER, args={'actor_name': USER, **publications.SMALL_LAYER.definition}, scope='class',)
+
+    def test_wfs_proxy(self, layer: Publication4Test, params):
+        with pytest.raises(gs_error) as raised_gs_error:
+            wfs_client.post_wfst(*layer,
+                                 operation=wfs_client.WfstOperation.INSERT,
+                                 version=wfs_client.WfstVersion.WFS20,
+                                 **params,
+                                 )
+        assert raised_gs_error.value.code == -1, f"{raised_gs_error=}, {raised_gs_error.code=}"
+        assert raised_gs_error.value.message == "WFS-T error", f"{raised_gs_error=}, {raised_gs_error.message=}"
+        assert raised_gs_error.value.data == {'status_code': 400}, f"{raised_gs_error=}, {raised_gs_error.data=}"
```

Comment:
I am surprised of many `f"..."` assert log messages. I think we should see good output even without them thanks to pytest.